### PR TITLE
0.10.1 types

### PR DESCRIPTION
**Pull request by mpetrunic**
- updated existing types
- added BeaconState object

I will try to migrate current beacon spec into openapi schema based on this branch

### CODE REVIEW

File: types/block.yaml [hunk 1]
```diff
@@ -1,52 +1,69 @@
 schemas:
+  BeaconBlockCommon:
```
Comment (djrtwo) [inline types/block.yaml]: My bad on bumping to v0.11.0 links without fixing params.

Is the plan here to get v0.10.x cleaned up and release. Then update to v0.11?

File: types/block.yaml [hunk 2]
```diff
@@ -1,52 +1,69 @@
...
+    # An abstract object to collect the common fields between the BeaconBlockHeader and the BeaconBlock objects
```
Comment (djrtwo) [inline types/block.yaml]: My bad on bumping to v0.11.0 links without fixing params.

Is the plan here to get v0.10.x cleaned up and release. Then update to v0.11?

File: types/misc.yaml [hunk 3]
```diff
@@ -1,33 +1,19 @@
...
+      epoch:
+        $ref: "./primitive.yaml#/schemas/Uint64"
+  Checkpoint:
+    type: object
+    description: "The [`Checkpoint`](https://github.com/ethereum/eth2.0-specs/blob/v0.10.1/specs/phase0/beacon-chain.md#checkpoint"
+    properties:
+      epoch:
+        $ref: "./primitive.yaml#/schemas/Uint64"
+      root:
+        $ref: "./primitive.yaml#/schemas/Root"
```
Comment (djrtwo) [inline types/misc.yaml]: github suggesting a newline here

File: types/operations/attestation.yaml [hunk 4]
```diff
@@ -1,94 +1,66 @@
...
-              - $ref: '../misc.yaml#/schemas/Hash'
-              - description: "Root of the previous crosslink."
-          data_root:
-            allOf:
-              - $ref: '../misc.yaml#/schemas/Hash'
-              - description: "Root of the crosslinked shard data since the previous crosslink."
+      source:
+        $ref: "../misc.yaml#/schemas/Checkpoint"
+      target:
+        $ref: "../misc.yaml#/schemas/Checkpoint"
```
Comment (djrtwo) [inline types/operations/attestation.yaml]: github suggesting newline

File: types/operations/deposit.yaml [hunk 5]
```diff
@@ -1,37 +1,34 @@
...
         allOf:
-          - $ref: '../misc.yaml#/schemas/Hash'
+          - $ref: '../primitive.yaml#/schemas/Root'
           - description: "The withdrawal credentials."
       amount:
-        type: integer
-        format: uint64
-        description: "Amount in Gwei."
+        allOf:
+          - $ref: "../primitive.yaml#/schemas/Uint64"
```
Comment (djrtwo) [inline types/operations/deposit.yaml]: Should we define a Gwei primitive?

File: types/primitive.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    description: "A string which uniquely identifies the client implementation and its version; similar to [HTTP User-Agent](https://tools.ietf.org/html/rfc7231#section-5.5.3)."
+    example: "Lighthouse / v0.1.5 (Linux x86_64)"
+
+  GenesisTime:
+    type: integer
+    format: uint64
+    description: "The genesis_time configured for the beacon node, which is the unix time at which the Eth2.0 chain began."
+    example: 1557716289
+
+  Uint64:
```
Comment (djrtwo) [inline types/primitive.yaml]: why `Uint64` defined as a string? whereas `GenesisTime` is defined as an integer

File: types/state.yaml [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        type: string
+        format: byte
+        pattern: "^0x[a-fA-F0-9]+$"
+        description: "Bit set for every recent justified epoch"
+      previous_justified_checkpoint:
+        $ref: "./misc.yaml#/schemas/Checkpoint"
+      current_justified_checkpoint:
+        $ref: "./misc.yaml#/schemas/Checkpoint"
+      finalized_checkpoint:
+        $ref: "./misc.yaml#/schemas/Checkpoint"
```
Comment (djrtwo) [inline types/state.yaml]: github suggesting newline

File: types/block.yaml (hunk 2 above)
Comment (mpetrunic) [reply types/block.yaml]: I though 0.10.x was going to be targeted version, I was wrong :smile: 
So yeah, I think it makes sense to update incrementally

File: types/operations/deposit.yaml (hunk 5 above)
Comment (mpetrunic) [reply types/operations/deposit.yaml]: yeah, we should!

File: types/primitive.yaml (hunk 6 above)
Comment (mpetrunic) [reply types/primitive.yaml]: It was left from initial version of api, will update to uint64

**Pull request merged**